test/GnosisSafeMultisig.js

```diff
@@ -132,5 +132,63 @@ describe("GnosisSafeMultisig", function () {
             }
             expect(await multisig.getThreshold()).to.equal(threshold);
         });
+
+        it("Setting a guard contract address that guards transactions for not being able to transfer ETH funds", async function () {
+            // Set up a guard contract with the last Safe owner being the guard owner
+            const SafeGuard = await ethers.getContractFactory("SafeGuard");
+            const safeGuard = await SafeGuard.deploy(signers[2].address);
+            await safeGuard.deployed();
+
+            const to = safeGuard.address;
+            const fallbackHandler = AddressZero;
+            const paymentToken = AddressZero;
+            const paymentReceiver = AddressZero;
+            const payment = 0;
+            let nonce = 0;
+            const payload = safeGuard.interface.encodeFunctionData("setGuardForSafe", [safeGuard.address]);
+            // Pack the data
+            const data = ethers.utils.solidityPack(["address", "address", "address", "address", "uint256", "uint256", "bytes"],
+                [to, fallbackHandler, paymentToken, paymentReceiver, payment, nonce, payload]);
+
+            // Create a multisig with the packed data
+            const tx = await gnosisSafeMultisig.create(defaultOwnerAddresses, threshold, data);
+            const result = await tx.wait();
+
+            // Check that the obtained address is not zero
+            expect(result.events[0].address).to.not.equal(AddressZero);
+            const proxyAddress = result.events[0].address;
+
+            // Get the safe multisig instance
+            const multisig = await ethers.getContractAt("GnosisSafe", proxyAddress);
+
+            // Check the multisig owners and threshold
+            const owners = await multisig.getOwners();
+            for (let i = 0; i < defaultOwnerAddresses.length; i++) {
+                expect(owners[i]).to.equal(defaultOwnerAddresses[i]);
+            }
+            expect(await multisig.getThreshold()).to.equal(threshold);
+
+            // Check the guard slot
+            const guardSlot = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
+            let slotValue = await ethers.provider.getStorageAt(multisig.address, guardSlot);
+            slotValue = ethers.utils.hexlify(ethers.utils.stripZeros(slotValue));
+            const safeGuardAddress = safeGuard.address.toLowerCase();
+            expect(slotValue).to.equal(safeGuardAddress);
```

Comment:
Checking that the create Safe instance has the correctly set guard slot with the required address.